### PR TITLE
added functionality to read the status of the start button

### PR DESCRIPTION
**Pull request by anton-firmware**
bees

### CODE REVIEW

File: robotd/devices.py [hunk 1]
```diff
@@ -159,11 +160,24 @@ def _set_power_outputs(self, level):
...
 
+    """
+    The following code was written by someone with a vague understanding of Python...
+    I'm sorry.
+    """
+    def read_the_damn_button(self):
+        result = self.device.control_read(64, 0, 8, 4)
+        status, = struct.unpack("i",  result)
+
+        if status == 0:
```
Comment (thomasleese) [inline robotd/devices.py]: We can change these lines to `return status != 0`.

File: robotd/devices.py [hunk 2]
```diff
@@ -159,11 +160,24 @@ def _set_power_outputs(self, level):
                 command,
             )
 
+    """
+    The following code was written by someone with a vague understanding of Python...
+    I'm sorry.
+    """
+    def read_the_damn_button(self):
```
Comment (thomasleese) [inline robotd/devices.py]: Might be better as:

```python
@property
def start_button_status(self):
```

File: robotd/devices.py [hunk 3]
```diff
@@ -159,11 +160,24 @@ def _set_power_outputs(self, level):
...
+            return False
+        else:
+            return True
+
     def make_safe(self):
         self._set_power_outputs(0)
 
     def status(self):
-        return {}
+        return {"start-button" : self.read_the_damn_button()}
```
Comment (thomasleese) [inline robotd/devices.py]: There shouldn't be a space on the left hand side of the colon.

```python
{'start-button': self.start_button_status}
```

**Review by thomasleese — COMMENTED** [review]
Looks good, got a few comments!

File: robotd/devices.py [hunk 4]
```diff
@@ -159,11 +160,24 @@ def _set_power_outputs(self, level):
                 command,
             )
 
+    """
+    The following code was written by someone with a vague understanding of Python...
```
Comment (Adimote) [inline robotd/devices.py]: Don't put yourself down! This code is good

File: robotd/devices.py (hunk 2 above)
Comment (Adimote) [reply robotd/devices.py]: yeah I'd avoid expletives in our code, some people could be offended

**Review by Adimote — CHANGES_REQUESTED** [review]
Basically fix what tom said, also has this code been tested with a real board?

File: robotd/devices.py (hunk 1 above)
Comment (anton-firmware) [reply robotd/devices.py]: all of them?

**Review by Adimote — APPROVED** [review]
LGTM, we should merge this once robotd is stable again (right now it's borked)

**Pull request merged**